src/ibek/sub_entity.py

```diff
@@ -0,0 +1,15 @@
+from pydantic import BaseModel, Field
+
+
+class SubEntity(BaseModel, extra="allow"):
+    """
+    A loosely defined class to declare the Entities
+    in an ibek.support.yaml file in the 'sub_entities' property of an Entity
+    section
+    """
```

Comment:
It would be good to have somie more detail here. Is this accurate?

```suggestion
class SubEntity(BaseModel, extra="allow"):
    """
    A partial or fully defined Entity nested inside an EntityDefinition

    This is a loosely defined class to declare nested Entities in an EntityDefinition
    that can be fully defined, or contain references to other Entities in an
    ibek.ioc.yaml file. References will be resolved after IOC deserialisation.

    These populate the 'sub_entities' property of an EntityDefinition in an
    ibek.support.yaml file.
    """
```